twilio/rest/__init__.py

```diff
@@ -116,11 +119,24 @@ def request(self, method, uri, params=None, data=None, headers=None, auth=None,
         if 'Accept' not in headers:
             headers['Accept'] = 'application/json'
 
-        if self.region:
-            head, tail = uri.split('.', 1)
-
-            if not tail.startswith(self.region):
-                uri = '.'.join([head, self.region, tail])
+        if self.region or self.edge:
+            pieces = uri.split('.')
+            prefix = pieces[0]
+            suffix = '.'.join(pieces[-2:])
```

Comment:
I'm concerned the `uri` that's provided may have a query path or fragment with a `.` in it. Because of this, better to assign the suffix to `'.'.join(pieces[2:])` or `'.'.join(pieces[3:])` in the below checks.

Reply:
Why wasn't this an issue/concern with twilio-node? also if there could be an unknown number of `.`s in the uri, or a custom domain, how can we determine whether a having more than 3 items in `pieces` indicates that there's a region/edge or whether there just happens to be a `.` elsewhere?

Reply:
Ah twilio-node uri goes through a uri parser